File: rust/89.gray-code.rs
/*
 * @lc app=leetcode id=89 lang=rust
 *
 * [89] Gray Code
 */

// @lc code=start
impl Solution {
    pub fn gray_code(n: i32) -> Vec<i32> {
        (0..(1 << n)).map(|x| x ^ (x >> 1)).collect()
    }
}
// @lc code=end

File: javascript/89.gray-code.js
/*
 * @lc app=leetcode id=89 lang=javascript
 *
 * [89] Gray Code
 */

// @lc code=start
/**
 * @param {number} n
 * @return {number[]}
 */
var grayCode = function (n) {
  const code = [];
  for (let i = 0; i < 1 << n; i++) {
    code.push(i ^ (i >> 1));
  }
  return code;
};
// @lc code=end
